cunyfirstapi/cunyfirstapi.py

```diff
@@ -91,12 +91,113 @@ def to_login(self):
         return self._session.get(constants.CUNY_FIRST_HOME_URL)
 
     def to_student_center(self):
-        response = self.session.get(constants.CUNY_FIRST_SIGNED_IN_STUDENT_CENTER_URL)
+        response = self._session.get(constants.CUNY_FIRST_SIGNED_IN_STUDENT_CENTER_URL)
         
         return response
 
-    def to_transcript_download(self):
-        pass
+    def to_transcript_download(self, college_code):
+        
+        # go to transcript request page without properly navigating, throw an error and get the values of the hidden form
+        url = constants.CUNY_FIRST_TRANSCRIPT_REQUEST_URL
+        response = self._session.get(url)
+        #print(r.text)
+        tree = html.fromstring(response.text)                  # parse the html DOM
+        data = {}                                       # store the key-value pairs for the form here
+        for el in tree.xpath('//input'):
+            # iterate through the hidden form
+            name = ''.join(el.xpath('./@name'))
+            value = ''.join(el.xpath('./@value'))
+            #print(name,value)
+            data[name] = value
+        
+        # manually make some changes to the form
+        data['ICAJAX'] = '1'
+        data['ICNAVTYPEDROPDOWN'] = '1'
+        data['ICYPos'] = '144'
+        data['ICStateNum'] = '1'
+        data['ICAction'] = 'DERIVED_SSS_SCR_SSS_LINK_ANCHOR4'
+        data['ICBcDomData'] = ''
+        data['DERIVED_SSS_SCL_SSS_MORE_ACADEMICS'] = '9999'
+        data['DERIVED_SSS_SCL_SSS_MORE_FINANCES'] = '9999'
+        data['CU_SF_SS_INS_WK_BUSINESS_UNIT'] = college_code
+        data['DERIVED_SSS_SCL_SSS_MORE_PROFILE'] = '9999'
+
+        # set url to student center menu
+        url = constants.CUNY_FIRST_SIGNED_IN_STUDENT_CENTER_URL
+        response = self._session.post(url = url, data = data)                 # post data to student center menu
+       
+        # navigate to the academics page
+        self._session.get(constants.CUNY_FIRST_MY_ACADEMICS_URL)
+        
+        # modify form for next stage
+        data['ICStateNum'] = '3'
+        data['ICAction'] = 'DERIVED_SSSACA2_SS_UNOFF_TRSC_LINK'
+        data['ICYPos'] = '95'
+        data['DERIVED_SSTSNAV_SSTS_MAIN_GOTO$7$'] = '9999'
+        data['DERIVED_SSTSNAV_SSTS_MAIN_GOTO$8$'] = '9999'
+
+        # go to transcript request page by posting data saying we want to go
+        response = self._session.post(url, data = data)
+
+        #data['url'] = url
+
+        url = constants.CUNY_FIRST_TRANSCRIPT_REQUEST_URL
+
+        response = self._session.get(url)            # actually go to transcript request page
+
+        return response
+
+    def download_transcript(self, college_code, data = None):
```

Comment:
Does this need to default to None or is the default None? Aka if you didnt have data=None would python allow you to write download_transcript("9090") or would it say it requires 2 parameters?

Reply:
need to default none explicitly otherwise python will require it as an argument